signalMachine.c

```diff
@@ -69,7 +69,7 @@ static inline char *makeReferenceKmer(const char *k_i, Strand strand, bool forwa
 }
 
 static inline char *kmerFromString(const char *string, int64_t start, int64_t kmerLength) {
-    char *k_i = st_malloc(kmerLength * sizeof(char));
+    char *k_i = st_malloc((kmerLength + 1) * sizeof(char));
```

Comment:
@tpesout Why did you add one here?

Reply:
you need kmerLength characters for the characters, plus an additional one for the null character to signal the end of the string (k_i[kmerLength] = '\0').  This was one of the buffer overflow bugs I found.

Reply:
I was actually thinking that but wasn't sure. Merging now!